src/network/jsonwebtask.cpp

```diff
@@ -57,44 +57,48 @@ QMimeType readContentType(
     return contentType;
 }
 
-bool readJsonContent(
+/// Returns std::nullopt on success and the raw payload otherwise.
+/// The JSON content is assigned to the output parameter jsonContent.
+std::optional<std::pair<QMimeType, QByteArray>> readJsonContent(
```

Comment:
That's odd std::optional<std::pair and auto in the using code. A recipe for maximum confusion. 
What are the alternatives? 

How about returning std::optional<CustomWebResponse> the readJsonContent can become a constructor or a createXXfromYY or something.

Reply:
This is just a helper function that is only used once. I am not introducing any wrapper. Both members of the pair are type-safe.

Reply:
I could reverse the return type and out-param to emphasize the regular control flow.

Reply:
Doesn't make it better, even more cumbersome to use. But a function is better than inlining the code, even if its signature is not suitable for a public API.

Reply:
Found a solution.